File: client/lib/playerStore.ts
import type { Player } from "./types";

const players: Player[] = [];

export function savePlayer(player: Player) {
  const idx = players.findIndex((p) => p.playerId === player.playerId);
  if (idx >= 0) {
    players[idx] = { ...player };
  } else {
    players.push({ ...player });
  }
}

export function getPlayers(): Player[] {
  return players;
}

/** Clears stored players (for tests). */
export function clearPlayers() {
  players.length = 0;
}

File: PromptWars_Aetherion_project/PromptWars_Aetherion/lib/player_store.py
from PromptWars_Aetherion.lib.types import Player

_players: list[Player] = []


def save_player(player: Player) -> None:
    for i, p in enumerate(_players):
        if p.playerId == player.playerId:
            _players[i] = player.model_copy()
            return
    _players.append(player.model_copy())


def get_players() -> list[Player]:
    return list(_players)


def clear_players() -> None:
    _players.clear()
